### PR TITLE
[#16] same name different messaging pattern allowed

### LINKED ISSUE #16: Service with the same name, but different messaging pattern shall be allowed.
## Brief feature description

Currently, when creating/opening a service with an existing name but a different messaging pattern one receives the failure that it either exists or that the messaging pattern does not match.

It would be better if this would be allowed and that a service is uniquely identified by the messaging pattern and the name.

## Detailed information

Should be easily implementable, just hash the messaging pattern as well to acquire the services uuid.

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Use either 'Closes #123' or 'Relates to #123' to reference the corresponding issue.

Closes #16

### CODE REVIEW

File: iceoryx2/src/service/static_config/mod.rs [hunk 1]
```diff
@@ -37,28 +37,39 @@ pub struct StaticConfig {
     pub(crate) messaging_pattern: MessagingPattern,
 }
 
+fn create_uuid<Hasher: Hash>(
+    service_name: &ServiceName,
+    messaging_pattern: &MessagingPattern,
+) -> Hasher {
+    let pattern_and_service = (<MessagingPattern as Into<u32>>::into(messaging_pattern.clone()))
```
Comment (elBoberido) [inline iceoryx2/src/service/static_config/mod.rs:44]: Is there a reason to not have the `MessagingPattern` implement `Copy`?

Just a general question, no need to change it in this PR if you agree to make it `Copy`.

Comment (elfenpiff) [reply iceoryx2/src/service/static_config/mod.rs:44]: I would add this in the last of the 3 PRs

Comment (elfenpiff) [reply iceoryx2/src/service/static_config/mod.rs:44]: @elBoberido no it is not possible since the static config contains a string which is not copyable.

Comment (elBoberido) [reply iceoryx2/src/service/static_config/mod.rs:44]: My bad. I assumed it was a simple number enum

**Pull request merged**